src/AuxDidAuth.ts

```diff
@@ -134,21 +138,35 @@ const signDidAuthInternal = async (
   hexPrivateKey: string,
   kid?: string
 ): Promise<string> => {
-  // assign specific JWT header
-  const header: JWTHeader = {
-    typ: "JWT",
-    alg: DidAuthKeyAlgorithm.ES256K,
-    kid: kid || `${issuer}#keys-1`,
-  };
+  let didkey = false;
+  if (
+    issuer.includes("did:key") ||
```

Comment:
you should check that it is `did:key:z6Mk` as described in https://w3c-ccg.github.io/did-method-key/#ed25519-x25519 because did:key supports different algorithms.

better use  a defaultAlgorithm=ES256K variable  instead of `didkey` boolean